risc0/zkvm/src/host/server/prove/tests.rs

```diff
@@ -580,6 +583,38 @@ mod docker {
         prover.prove(env, MULTI_TEST_ELF).unwrap().receipt
     }
 
+    fn exec_verify(receipt: &Receipt) {
+        let input: (Receipt, Digest) = (receipt.clone(), MULTI_TEST_ID.into());
+        let env = ExecutorEnv::builder()
+            .write(&input)
+            .unwrap()
+            .build()
+            .unwrap();
+        let session = ExecutorImpl::from_elf(env, VERIFY_ELF)
+            .unwrap()
+            .run()
+            .unwrap();
+        assert_eq!(session.exit_code, ExitCode::Halted(0));
+        println!("{:?}", session.stats());
+    }
+
+    #[test]
+    fn verify_in_guest() {
+        let composite_receipt = generate_receipt(ProverOpts::composite());
+        exec_verify(&composite_receipt);
+        let succinct_receipt = get_prover_server(&ProverOpts::succinct())
+            .unwrap()
+            .compress(&ProverOpts::succinct(), &composite_receipt)
+            .unwrap();
+        exec_verify(&succinct_receipt);
+        let compact_receipt = get_prover_server(&ProverOpts::compact())
+            .unwrap()
+            .compress(&ProverOpts::compact(), &succinct_receipt)
+            .unwrap();
+        exec_verify(&compact_receipt);
+        compact_receipt.inner.compact().unwrap();
```

Comment:
I'm surprised this works, I would have guessed that compact receipts can't be verified in the guest right now.

Reply:
We have the groth16 verify as an example so I thought that it was feasible but I didn't know it would be this easy. Thanks @capossele!